Remotion/Mixins/UnitTests/Core/CodeGeneration/AttributeBasedMetadataImporterTest.cs

```diff
@@ -180,63 +180,46 @@ public void GetMetadataForMixinType_NoAttribute ()
     public void GetMetadataForMixedType ()
     {
       var classContext1 = ClassContextObjectMother.Create(typeof (object));
+      var type = typeof (ClassWithConcreteMixedTypeAttribute);
+      var importer = new AttributeBasedMetadataImporter ();
 
-      var attribute1 = ConcreteMixedTypeAttribute.FromClassContext (classContext1, new Type[0]);
-
-      var typeMock = MockRepository.GenerateMock<_Type> ();
-      typeMock.Expect (mock => mock.GetCustomAttributes (typeof (ConcreteMixedTypeAttribute), false)).Return (new[] { attribute1 });
-      typeMock.Replay ();
+      var result = importer.GetMetadataForMixedType (type);
 
-      var importer = new AttributeBasedMetadataImporter ();
-      var result = importer.GetMetadataForMixedType (typeMock);
       Assert.That (result, Is.EqualTo (classContext1));
-
-      typeMock.VerifyAllExpectations ();
     }
 
     [Test]
     public void GetMetadataForMixedType_NoAttribute ()
     {
-      var typeMock = MockRepository.GenerateMock<_Type> ();
-      typeMock.Expect (mock => mock.GetCustomAttributes (typeof (ConcreteMixedTypeAttribute), false)).Return (new ConcreteMixedTypeAttribute[0]);
-      typeMock.Replay ();
-
+      var type = typeof (ClassWithNoAttributes);
       var importer = new AttributeBasedMetadataImporter ();
-      var result = importer.GetMetadataForMixedType (typeMock);
-      Assert.That (result, Is.Null);
 
-      typeMock.VerifyAllExpectations ();
+      var result = importer.GetMetadataForMixedType (type);
+
+      Assert.That (result, Is.Null);
     }
 
     [Test]
     public void GetIdentifierForMixinType ()
     {
-      var typeMock = MockRepository.GenerateMock<_Type> ();
       var identifier = new ConcreteMixinTypeIdentifier (typeof (object), new HashSet<MethodInfo>(), new HashSet<MethodInfo>());
-      var attribute = ConcreteMixinTypeAttribute.Create (identifier);
+      var type = typeof (ClassWithConcreteMixinTypeAttribute);
+      var importer = new AttributeBasedMetadataImporter ();
 
-      typeMock.Expect (mock => mock.GetCustomAttributes (typeof (ConcreteMixinTypeAttribute), false)).Return (new[] { attribute });
```

Comment:
Given that the test is asserting the attribute inheritance via the expectation, we should add a test that uses a derived type without attribute to verify that we don't inherit the attribute. Yes, the attribute is already defined as `Inherited=false`, but it may be that this code here is the only place that verifies the correct behavior. I would probably also accept a pointer to another piece of code that deals with inheritance, but I think it's quicker to just add a test here for the two attribute types.